### DISCUSSION TITLE
gRPC web with Vercel

### DISCUSSION THREAD

**Discussion by lucksp** [grpc/grpc-web#1161] [Q&A - General]
We are hosting our web app on Vercel since it's a React app with NextJS.  Does this server have any limitation implementing gRPC web?  It's just a browser based network request, so it shouldn't...but I assume we still need a proxy?   We don't need a node or docker image on our server, correct?

**Comment by sampajano** [comment]
Thanks for the question! :)

The gRPC-Web JS library is designed to run inside a browser, and would probably not work if you run it for server-side rendering (if i understand correctly about what Next.js does) -- mainly because we rely on XHR and doesn't support [fetch()](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API) yet (FR at https://github.com/grpc/grpc-web/issues/1134). We also don't yet support Service Workers for the same reason.

But if your question is regarding running a gRPC backend on your Next.js server, i don't have much idea :) But yes, if you do that you'll probably need to run Envoy to proxy gRPC-Web traffic from the browser.

For server use cases, maybe [gRPC-node](https://github.com/grpc/grpc-node) or [gRPC-js](https://github.com/grpc/grpc-node/tree/master/packages/grpc-js) could work for you, but i'm not an expert. (CC'ing @murgatroid99 in case he has some thoughts on that :))

Hope that helps.. :)
